syntax-highlighting-code-block.php

```diff
@@ -271,8 +278,12 @@ function render_block( $attributes, $content ) {
 		}
 
 		$highlighter = new \Highlight\Highlighter();
-		$language    = $attributes['language'];
-		$code        = html_entity_decode( $matches['code'], ENT_QUOTES );
+		if ( ! empty( $auto_detect_languages ) ) {
+			$highlighter->setAutodetectLanguages( $auto_detect_languages );
+		}
```

Comment:
This is good! Without this check and setting autodetect to an empty array would still work, however, it would trigger the highlighter to reindex all of its languages.

Reply:
That's right..

What is the behavior when the array is empty? Does it iterate over all languages in alphabetical order? If so, perhaps a default should be devised based on how popular the languages are?

Reply:
When the highlighter is configured to have 0 auto-detect languages, either by calling `setAutodetectLanguages` explicitly or setting the subset in `highlightAuto`, it will use *every* language registered (all ~185 languages) via brute-force.

Reply:
As for having a default set of languages to autodetect:

- if you mean in highlight.php, that'd be left to highlight.js to decide
- but if you mean for this plug-in, that's entirely feasible

Reply:
> When the highlighter is configured to have 0 auto-detect languages, either by calling `setAutodetectLanguages` explicitly or setting the subset to null in `highlightAuto`, it will use _every_ language registered (all ~185 languages) via brute-force.

And this brute force of all 185 languages is the default behavior?

> As for having a default set of languages to autodetect:

Yeah, I suppose in this plugin. If we could somehow determine the relative popularity of each language by frequency of use on GitHub, for example, then this could be the default value used here when calling `setAutodetectLanguages`. Users could filter it to modify the order.

Reply:
> And this brute force of all 185 languages is the default behavior?

So the default behavior for highlight.php is to use [`"xml", "json", "javascript", "css", "php", "http"`](https://github.com/scrivo/highlight.php/blob/master/Highlight/Highlighter.php#L97-L105) as the default auto-detect languages. However, this is legacy behavior and _will_ change since [highlight.js does not have any default languages](https://github.com/highlightjs/highlight.js/blob/master/src/highlight.js#L574) (I should clarify this in the README). highlight.js will [brute force its way through all its registered languages](https://github.com/highlightjs/highlight.js/blob/3f2b813a3732f45b0a5e719e3d16fa55bf73c5f9/src/highlight.js#L35-L44) but it has different default builds with just *some* languages, so the brute force approach doesn't typically go through all 185 languages unless you use a full build. highlight.php only has a full build.

But yes, if you give the autodetection an empty array, the default behavior will be to use all 185 languages.